### PR TITLE
Update to Vert.x 4 EDGCOMMON-30

### PR DESCRIPTION
**Pull request by adamdickmeiss**
Based on EDGCOMMON-28 PR (that one should be merged before this one).

This PR will not be merged until https://github.com/folio-org/edge-common/pull/39 is merged. But comments are of course welcomed.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -250,7 +254,7 @@
       <dependency>
         <groupId>io.vertx</groupId>
         <artifactId>vertx-stack-depchain</artifactId>
-        <version>3.5.4</version>
+        <version>4.0.0</version>
```
Comment (julianladisch) [inline pom.xml:257]: Inconsistent jackson versions 2.9.6 and 2.11.2:
```
mvn dependency:tree -Dincludes=*jackson*

[INFO] org.folio:edge-common:jar:3.1.0-SNAPSHOT
[INFO] +- io.vertx:vertx-core:jar:4.0.0:compile
[INFO] |  \- com.fasterxml.jackson.core:jackson-core:jar:2.11.3:compile
[INFO] +- com.fasterxml.jackson.core:jackson-annotations:jar:2.11.3:compile
[INFO] +- com.fasterxml.jackson.dataformat:jackson-dataformat-xml:jar:2.9.6:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.11.3:compile
[INFO] |  \- com.fasterxml.jackson.module:jackson-module-jaxb-annotations:jar:2.9.6:compile
[INFO] \- com.amazonaws:aws-java-sdk-ssm:jar:1.11.313:compile
[INFO]    \- com.amazonaws:aws-java-sdk-core:jar:1.11.313:compile
[INFO]       \- com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:jar:2.11.3:compile
```

```suggestion
        <version>4.0.0</version>
        <!-- Update jackson-bom accordingly: https://github.com/vert-x3/vertx-dependencies/blob/master/pom.xml -->
```

File: src/main/java/org/folio/edge/core/EdgeVerticle.java [hunk 2]
```diff
@@ -51,7 +51,7 @@
   public final int port;
   public final String okapiURL;
   public final String apiKeySources;
-  public final long reqTimeoutMs;
+  public final int reqTimeoutMs;
```
Comment (julianladisch) [inline src/main/java/org/folio/edge/core/EdgeVerticle.java:54]: https://vertx.io/docs/apidocs/io/vertx/core/http/HttpClientRequest.html#setTimeout-long- takes a long, OkapiClient takes a long, why do you change it to int?

File: src/main/java/org/folio/edge/core/EdgeVerticle.java [hunk 3]
```diff
@@ -106,19 +106,25 @@ protected EdgeVerticle() {
...
-      if (result.succeeded()) {
-        future.complete();
-      } else {
-        future.fail(result.cause());
-      }
-    });
+  public void start(Promise<Void> promise) {
+    server.requestHandler(router).listen(port)
+        .onSuccess(res -> promise.complete())
+        .onFailure(promise::fail);
```
Comment (julianladisch) [inline src/main/java/org/folio/edge/core/EdgeVerticle.java:112]: server.requestHandler(router)
        .listen(port)
        .onComplete(promise);

File: src/main/java/org/folio/edge/core/EdgeVerticle2.java [hunk 4]
```diff
@@ -87,38 +84,15 @@ public void start(Future<Void> future) {
...
     final Router router = defineRoutes();
 
-    server.requestHandler(router::accept).listen(port, result -> {
-      if (result.succeeded()) {
-        future.complete();
-      } else {
-        future.fail(result.cause());
-      }
-    });
+    server.requestHandler(router).listen(port)
```
Comment (julianladisch) [inline src/main/java/org/folio/edge/core/EdgeVerticle2.java:87]: server.requestHandler(router).listen(port).onComplete(promise);

File: src/main/java/org/folio/edge/core/utils/OkapiClient.java [hunk 5]
```diff
@@ -152,26 +163,30 @@ public void post(String url, String tenant, String payload, MultiMap headers,
           request.headers().get(X_OKAPI_TOKEN)));
     }
 
-    request.handler(responseHandler)
-      .exceptionHandler(exceptionHandler)
-      .setTimeout(reqTimeout);
+    // TODO   .setTimeout(reqTimeout);
```
Comment (julianladisch) [inline src/main/java/org/folio/edge/core/utils/OkapiClient.java:166]: setConnectTimeout?

File: src/main/java/org/folio/edge/core/EdgeVerticle.java (hunk 3 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/edge/core/EdgeVerticle.java:112]: doesn't compile Provided `Promise<Void` . Required `Handler<AsyncResult<HttpServer>>`.

File: src/main/java/org/folio/edge/core/EdgeVerticle2.java (hunk 4 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/edge/core/EdgeVerticle2.java:87]: Same here. That construct does not compile.

File: src/main/java/org/folio/edge/core/utils/OkapiClient.java (hunk 5 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/edge/core/utils/OkapiClient.java:166]: Forgot to remove that comment.  I have added setConnectTimeout (already added setIdleTimeout).

File: pom.xml (hunk 1 above)
Comment (adamdickmeiss) [reply pom.xml:258]: Good catch. I'll update junit as well.

File: src/main/java/org/folio/edge/core/EdgeVerticle.java (hunk 3 above)
Comment (julianladisch) [reply src/main/java/org/folio/edge/core/EdgeVerticle.java:112]:
```
server.requestHandler(router)
    .listen(port)
    .<Void>mapEmpty()
    .onComplete(promise);
```

File: src/main/java/org/folio/edge/core/EdgeVerticle2.java (hunk 4 above)
Comment (julianladisch) [reply src/main/java/org/folio/edge/core/EdgeVerticle2.java:87]:
```
    server.requestHandler(router)
    .listen(port)
    .<Void>mapEmpty()
    .onComplete(promise);
```

File: pom.xml (hunk 1 above)
Comment (julianladisch) [reply pom.xml:258]: You fixed it, thanks, can be closed.

File: src/main/java/org/folio/edge/core/utils/OkapiClient.java (hunk 5 above)
Comment (julianladisch) [reply src/main/java/org/folio/edge/core/utils/OkapiClient.java]: You fixed it, thanks, can be closed.

File: src/main/java/org/folio/edge/core/EdgeVerticle2.java (hunk 4 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/edge/core/EdgeVerticle2.java:87]: ok

File: src/main/java/org/folio/edge/core/EdgeVerticle.java (hunk 3 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/edge/core/EdgeVerticle.java:112]: ok

**Review by julianladisch — APPROVED** [review]
Looks good, thanks!

**Comment by craigmcnally** [comment]
As discussed, EdgeVerticle should be deprecated in v3.1 and removed altogether in v4.  Going forward EdgeVerticle2 should be used.

**Comment by mreno-EBSCO** [comment]
Can we rename `EdgeVerticle2` to `EdgeVerticle` in v4.0.0? I think that will be less confusing to new edge-common users.

**Comment by adamdickmeiss** [comment]
I'll wait a bit on merging this after 3.1 release. When this PR is merged, I'll make another PR where the EdgeVerticle2 becomes EdgeVerticle.

**Comment by adamdickmeiss** [comment]
SQ complains about use of deprecated EdgeVerticle .. from other PR.. This will go away when remove deprecated EdgeVerticle.

**Pull request merged**